felix/bpf/maps/syscall.go

```diff
@@ -201,15 +202,16 @@ func NewIterator(mapFD FD, keySize, valueSize, maxEntries int) (*Iterator, error
 	valueStride := align64(valueSize)
 
 	m := &Iterator{
-		mapFD:         mapFD,
-		maxEntries:    maxEntries,
-		keySize:       keySize,
-		valueSize:     valueSize,
-		keyStride:     keyStride,
-		valueStride:   valueStride,
-		keysBufSize:   keyStride * IteratorNumKeys,
-		valuesBufSize: valueStride * IteratorNumKeys,
-		keysValues:    make(chan keysValues),
+		mapFD:                mapFD,
+		maxEntries:           maxEntries,
+		keySize:              keySize,
+		valueSize:            valueSize,
+		keyStride:            keyStride,
+		valueStride:          valueStride,
+		keysBufSize:          keyStride * IteratorNumKeys,
+		valuesBufSize:        valueStride * IteratorNumKeys,
+		keysValues:           make(chan keysValues),
+		batchLookupSupported: isBatchOpsSupported,
```

Comment:
since `isBatchOpsSupported()` is a `sync.OnceValue`, wouldn't it work to call it here when creating the Iterator instead of passing it as an arg? just thinking of keeping the API cleaner, but let me know if there's a reason it wouldn't work...

Reply:
I thought of this. But this way gives us a chance to test as we don't run tests on any kernel which doesn't support batch operations.